01-test

```diff
@@ -2,38 +2,50 @@
 
 # Unit testing
 
-cargo test \
-    && echo Cargo test succeeded. \
-    || ( echo Cargo test failed. && exit 1 )
+if cargo test
+then
+  echo Cargo test succeeded.
+else
+  ( echo Cargo test failed. && exit 1 )
+fi
 
 
 # Functional tests
 
-cd ./tests/functional/
+cd ./tests/functional/ || exit
```

Comment:
It is better here not to exit silently. If this folder does not exist, there is really something wrong going on and the script should make a lot of noise.